packages/scss/src/components/_button.scss

```diff
@@ -227,40 +213,43 @@
 			top: 0;
 			vertical-align: middle;
 		}
+	}
+}
 
-		&.mod-smaller {
-			&::after {
-				font-size: 1rem;
-			}
-		}
-
-		&.mod-small {
-			&::after {
-				font-size: 1.2rem;
-			}
+// sizing for states
+.button.is-loading {
```

Comment:
In the last refacto, I simplified it by putting the sizing loop inside each mod. But indeed, having a big sizing loop which include each mod might be a good idea.